### DISCUSSION TITLE
Unable to send stream after reconnecting to a room in Ant Media Server iOS SDK

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5249] [FAQs]
We have a scenario where two devices, Device A (iOS-SDK) and Device B (Web), join a room and start streaming audio to each other. If Device A is turned off and loses the connection, it's expected that the connection is lost, while the room remains active with Device B still present.

When Device A reconnects to the same room after being turned back on, it can successfully receive streams from Device B. However, we have observed that Device A is unable to send its own stream.

We are using the Ant Media Server iOS SDK for Device A and the Web SDK for Device B.

We would like to understand the possible reasons behind this issue and explore any solutions or workarounds that would allow Device A to resume sending its stream after reconnecting to the room.

1) How iOS SDK notifies the client app about the state of reconnection?
2) Once Reconnection is established, how SDK notifies the client app about it?
3) Which delegate of AntMediaClient/WebRTCClient should we rely on to show/hide the "Reconnection..." status and replay/republish streams?

**Comment by mekya** [comment]
I've specifically tested this scenario and I confirm that there [was a issue](https://github.com/ant-media/Ant-Media-Server/issues/5258) in reconnection that has been [fixed in this PR](https://github.com/ant-media/WebRTC-iOS-SDK/pull/24)

I specifically tested the following scenario in my version (multitrack-conference.html)

1. Device A(iOS-SDK) join to the room
2. Device B(Web) join to the room
3. Device A and Device B see and hear each other
4. Close network connection for Device A
5. After the disconection timeout, Device A(iOS-SDK) drops in the room completely
6. Enable the network for Device A again 

I've checked it's publishing and playing as expected. 

If you still have the problem or you want me to make demo, I'm more than happy to do that.

Cheers
Oz

**Reply by FarhodKurbonov** [reply]
I guess issue was resolved on SDK.  But I repeat our  initial questions again:

1) How iOS SDK **notifies** client app about the state of reconnection?
2) Once Reconnection is established, how SDK **notifies** client app about it?
3) **Which delegate** of AntMediaClient/WebRTCClient should we rely on to show/hide "Reconnection..." status and replay/republish streams?

**Reply by mekya** [reply]
Hi @Farhod,
Thank you for your question. I've missed that. 

> 1.How iOS SDK notifies client app about the state of reconnection?
If WebRTC connection is dropped due to network, sdk [triggers `disconnected` delegate method](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTCiOSSDK/api/AntMediaClientDelegate.swift#L86 ) . 

Server also detects the disconnection after the timeout(default 10 secs) and sends `publish finished` message through websocket and it triggers the [`publishFinished`](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTCiOSSDK/api/AntMediaClientDelegate.swift#L81) delegate method. There is also a [`play finished` message/delegate method](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTCiOSSDK/api/AntMediaClientDelegate.swift#L69) if you are playing a stream 

If you want to get notified for all connection status changes detected by the client app, we can send all connection state changes do the delegate method. 

> 2. Once Reconnection is established, how SDK notifies client app about it?
After that when publishing is started, it triggers [`publishStarted`](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTCiOSSDK/api/AntMediaClientDelegate.swift#L75)

If you are playing a stream, it triggers [`playStarted` delegate method](https://github.com/ant-media/WebRTC-iOS-SDK/blob/master/WebRTCiOSSDK/api/AntMediaClientDelegate.swift#L63)

3. Which delegate of AntMediaClient/WebRTCClient should we rely on to show/hide "Reconnection..." status and replay/republish streams?

When the `disconnected` delegate method is called and if the user is not requested to stop publishing/playing , it's time to show "Reconnection ...". 
When you receive `publishStarted` or `playStarted` delegate, it's time to hide "Reconnection ..."


I hope these are answers for your question. Please let me know if you have further questions or I can help with anything. 

Have a great weekend

**Reply by KavyashreeBasavesh** [reply]
Hi, 
 We are facing the same issue with the Android SDK. Is there any possible way you can help me out?

**Reply by mekya** [reply]
Hi @KavyashreeBasavesh,

Definitely we can help you out. Let's schedule a meeting on consultancy section as showed in the image below on [antmedia.io](https://antmedia.io/#ribbon)

<img width="929" alt="Screenshot 2023-08-14 at 10 38 13" src="https://github.com/ant-media/Ant-Media-Server/assets/3456251/7c2ef8f5-78d2-466c-8ffa-bd9752fb5f1f">


Cheers
